### PR TITLE
Use current git branch to set various release parameters

### PR DESCRIPTION
**Pull request by olblak**
~~This pull request simplifies how we set JENKINS_VERSION, RELEASE_PROFILE, RELEASELINE based on git branch~~

For this I am also improving `getJenkinsVersion.py` to search for the latest version based on a pattern like `2.235` then we sort the versions found in alphabetical order.

This PR also improves 'getJenkinsVersion.py' in order to retrieve the latest version available based on an identifer instead of doing an alphabetical sort 

Rename release.sh to release.bash in order to use it from BATS

### CODE REVIEW

File: utils/release.sh [hunk 1]
```diff
@@ -296,6 +297,63 @@ cat <<EOT> settings-release.xml
 EOT
 }
 
+# guessGitBranchInformation tries to guesss PROFILE, RELEASELINE, and JENKINS_VERSION based on the git branch
```
Comment (timja) [inline utils/release.sh:300]:
```suggestion
# guessGitBranchInformation tries to guess PROFILE, RELEASELINE, and JENKINS_VERSION based on the git branch
```

**Comment by oleg-nenashev** [comment]
Putting on-hold as requiested by @olblak

**Comment by olblak** [comment]
We are definitely missing some unit test here, especially for the new bash function 'guessGitBranchInformation'

**Comment by olblak** [comment]
We can run `make test` to run the bats tests, I am still wondering what which branch name we would use for the security weekly release

File: README.adoc [hunk 2]
```diff
@@ -69,9 +69,8 @@ The release process implies following Git repository:
 
 It accepts following values:
 
-- `weekly`, fetches the latest version available on a maven repository based on the maven-metada.xml.
-- `stable`, fetches the latest version that matches pattern X.Y.Z.
-- `version`, a valid version.
+- `latest`, fetches the latest version available on a maven repository based on the maven-metada.xml.
```
Comment (timja) [inline README.adoc:72]:
```suggestion
- `latest`, fetches the latest version available in a maven repository based on the maven-metada.xml.
```

File: utils/release.bats [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    echo "EXPECTED JENKINS_VERSION = $EXPECTED_JENKINS_VERSION"
+    echo "GOT JENKINS_VERSION = $JENKINS_VERSION"
+    [ "$JENKINS_VERSION" = "$EXPECTED_JENKINS_VERSION" ]
+  }
+}
+
+testGuessGitBranchInformation "stable-2.235" "stable" "-stable" "2.235"
+testGuessGitBranchInformation "master" "weekly" "" "latest"
+testGuessGitBranchInformation "security-stable-2.235" "security" "-stable" "2.235"
+#testGuessGitBranchInformation "security-2.235" "security" "" "latest"
```
Comment (timja) [inline utils/release.bats:41]: why's this commented out?

Comment (olblak) [reply utils/release.bats:41]: "I am still wondering which branch name we would use for the security weekly release"

Comment (olblak) [reply utils/release.bats:41]: Depending on that, I'll adapt the test

Comment (olblak) [reply utils/release.bats:41]: Right now a weekly security release branch name has a specifc code name so I can't guess any information based on that

**Comment by olblak** [comment]
I think this PR could be merged, while the initial motivation hasn't been met, which was to remove the need to define JENKINS_VERSION, RELEASE_PROFILE, RELEASELINE. Overall it remains a good improvements, the python code that guess the latest version is more robust and has unittests, while the function `guessGitBranchInformation` is almost ready, it doesn't cover weekly security releases so I am keeping it as a fallback situation

**Review by timja — APPROVED** [review]

**Comment by daniel-beck** [comment]
It's unclear to me what the purpose of this PR now is (the description seems outdated), and some changes don't make explicit why they exist. I don't feel qualified to review this.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,15 @@
+.PHONY: test
+
+test:
+	docker run \
+		-v /home/olblak/Project/Jenkins-infra/release:/bats \
```
Comment (olblak) [inline Makefile:5]:
```suggestion
		-v "$PWD:/bats" \
```

**Pull request merged**